app/src/main/java/com/github/se/cyrcle/ui/review/ReviewScreen.kt

```diff
@@ -0,0 +1,125 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.cyrcle.model.parking.ParkingViewModel
+import com.github.se.cyrcle.model.review.ReviewViewModel
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.navigation.Screen
+import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ReviewScreen(
+    navigationActions: NavigationActions,
+    parkingViewModel: ParkingViewModel,
+    reviewViewModel: ReviewViewModel
+) {
+  var sliderValue by remember { mutableStateOf(0f) }
+  var textValue by remember { mutableStateOf("") }
+
+  val context = LocalContext.current // Get the current context
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            navigationActions = navigationActions,
+            title = {
+              Text(
+                  text = "Add Your Review",
+                  fontSize = 20.sp,
+                  fontWeight = FontWeight.Bold,
+                  color = MaterialTheme.colorScheme.secondary,
+                  modifier = Modifier.testTag("TopAppBarTitle"))
+            })
+      }) { paddingValues ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(paddingValues).padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center) {
+              // Stars reflecting the slider value (supports half stars)
+              Row(
+                  modifier = Modifier.padding(8.dp).testTag("StarsRow"),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    for (i in 1..5) {
+                      val starIcon =
+                          when {
+                            i <= sliderValue -> Icons.Filled.Star // Full star
+                            i - 0.5f == sliderValue -> Icons.Filled.StarHalf // Half star
+                            else -> Icons.Filled.StarBorder // Empty star
+                          }
+                      Icon(
+                          imageVector = starIcon,
+                          contentDescription = null,
+                          tint = MaterialTheme.colorScheme.primary,
+                          modifier = Modifier.size(40.dp).testTag("Star$i"))
+                    }
+                  }
+
+              // Display the experience text based on the slider value
+              Text(
+                  text =
+                      when (sliderValue) {
+                        in 0f..1f -> "Terrible experience!"
+                        in 1f..2f -> "Bad experience :("
+                        in 2f..3f -> "Poor experience :/"
+                        3f -> "Average experience..."
+                        in 3.5f..4f -> "Good experience!"
+                        in 4.5f..5f -> "Great experience!"
+                        else -> ""
+                      },
```

Comment:
I think that having non-linear opinions (i.e. the interval is not the same for each line) is bad for the user (cognitive charge)

Reply:
I agree with this actually! Was not sure where to draw the line as "Average" and UI-wise it didn't seem too jarring so I didn't change it yet

Reply:
I would say that either of the following scheme would be valid
- 1, 1.5, 2, ... : but the texts might need to be more confusing than helpful
- 1-2, 2.5-3.5, 4-5 : Three steps, might be too simple
- 1-1.5, 2-2.5, ... : Five steps, might be the best

Reply:
I like this a lot, this will be a future design choice we'll implement in a future sprint